logging/bin/security_create_rbac.sh

```diff
@@ -1,41 +1,56 @@
+# Copyright © 2021, 2020, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
 #
-# Creates the following RBAC structures
+# Creates the following RBAC structures (NST=NAMESPACE|NAMESPACE_TENANT):
 #
-#                                   /-- [ROLE: kibana_user]       (allows access to Kibana)
-# [BACKEND_ROLE: {NS}_kibana_user]<-
-#                                   \-- [ROLE: search_index_{NS}] (allows access to log messages from {NS})
+#
+#                                    /-- [ROLE: kibana_user]       (allows access to Kibana)
+# [BACKEND_ROLE: {NST}_kibana_user]<-
+#                                    \-- [ROLE: search_index_{NST}] (allows access to log messages from {NST})
 #
 #
 #
 # READONLY ROLE
 #
-#                                        /- [ROLE: cluster_ro_perms]  (limits access to cluster to read-only)
-#                                       /-- [ROLE: kibana_read_only]  (limits Kibana access to read-only)
-#                                      /--- [ROLE: kibana_user]       (allows access to Kibana)
-# [BACKEND_ROLE: {NS}_kibana_ro_user]<-
-#                                      \--- [ROLE: search_index_{NS}] (allows access to log messages from {NS})
+#                                         /- [ROLE: cluster_ro_perms]  (limits access to cluster to read-only)
+#                                        /-- [ROLE: kibana_read_only]  (limits Kibana access to read-only)
+#                                       /--- [ROLE: kibana_user]       (allows access to Kibana)
+# [BACKEND_ROLE: {NST}_kibana_ro_user]<-
+#                                       \--- [ROLE: search_index_{NST}] (allows access to log messages from {NST})
 #
 
 cd "$(dirname $BASH_SOURCE)/../.."
 source logging/bin/common.sh
 this_script=`basename "$0"`
 
+source logging/bin/rbac-include.sh
+source logging/bin/apiaccess-include.sh
+
+
 function show_usage {
-  log_info  "Usage: $this_script NAMESPACE"
-  log_info  ""
-  log_info  "Creates access control artifacts (e.g. roles, role-mappings, etc.) to limit access to the specified namespace."
-  log_info  ""
-  log_info  "        NAMESPACE - (Required) The Viya deployment/Kubernetes Namespace for which access controls should be created"
+  log_message  "Usage: $this_script NAMESPACE [TENANT]"
+  log_message  ""
+  log_message  "Creates access controls (e.g. roles, role-mappings, etc.) to limit access to the specified namespace and, optionally, the specified tenant within that namespace."
+  log_message  ""
+  log_message  "        NAMESPACE - (Required) The Viya deployment/Kubernetes Namespace for which access controls should be created"
+  log_message  ""
+  log_message  "        TENANT    - (Optional) The tenant with the Viya deployment/Kubernetes Namespace for which access controls should be created"
+  log_message  ""
 }
 
+#TO DO: Move to named args
+
 NAMESPACE=${1}
-READONLY=${2:-false}
+TENANT=${2}
+READONLY=${3:-false}
 
+if [ "$TENANT" == "--add_read_only" ]; then
```

Comment:
Not really...we're processing the args one by one since they're positional.  Tenant is optional, so it may or may not be there.  If it isn't specified (e.g. security_create_rbac.sh myNamespace  --add_read_only), then the read only flag ends up as the 2nd arg, so I have to check for that.  In a future release, we'll move to named args and this will go away.